splunk_eventgen/lib/raterplugin.py

```diff
@@ -0,0 +1,175 @@
+                logger.debug(
+                    "minuteOfHourRate for sample '%s' in app '%s' is %s" % (self.sample.name, self.sample.app, rate))
+                rateFactor *= rate
+            except KeyError:
+                import traceback
+                stack = traceback.format_exc()
+                logger.error(
+                    "Minute of hour rate failed for sample '%s'.  Stacktrace %s" % (self.sample.name, stack))
+        if type(self.sample.dayOfMonthRate) == dict:
+            try:
+                rate = self.sample.dayOfMonthRate[str(self.sample.now().day)]
+                logger.debug(
+                    "dayOfMonthRate for sample '%s' in app '%s' is %s" % (self.sample.name, self.sample.app, rate))
+                rateFactor *= rate
+            except KeyError:
+                import traceback
+                stack = traceback.format_exc()
+                logger.error(
+                    "Day of Month rate for sample '%s' failed.  Stacktrace %s" % (self.sample.name, stack))
+        if type(self.sample.monthOfYearRate) == dict:
+            try:
+                rate = self.sample.monthOfYearRate[str(self.sample.now().month)]
+                logger.debug(
+                    "monthOfYearRate for sample '%s' in app '%s' is %s" % (self.sample.name, self.sample.app, rate))
+                rateFactor *= rate
+            except KeyError:
+                import traceback
+                stack = traceback.format_exc()
+                logger.error(
+                    "Month Of Year rate failed for sample '%s'.  Stacktrace %s" % (self.sample.name, stack))
+        return rateFactor
+
+    def single_queue_it(self, count):
+        """
+        This method is used for specifying how to queue your rater plugin based on single process
+        :param count:
+        :return:
+        """
+        et = self.sample.earliestTime()
+        lt = self.sample.latestTime()
+        if count < 1 and count != -1:
+            logger.info(
+                "There is no data to be generated in worker {0} because the count is {1}.".format(
+                    self.sample.config.generatorWorkers, count))
+        else:
+            genPlugin = self.generatorPlugin(sample=self.sample)
+            # Adjust queue for threading mode
+            genPlugin.updateConfig(config=self.config, outqueue=self.outputQueue)
+            genPlugin.updateCounts(count=count, start_time=et, end_time=lt)
+            try:
+                self.generatorQueue.put(genPlugin)
+                logger.info(("Put {0} MB of events in queue for sample '{1}'" +
+                                  "with et '{2}' and lt '{3}'").format(
+                                      round((count / 1024.0 / 1024), 4),
+                                      self.sample.name, et, lt))
+            except Full:
+                logger.warning("Generator Queue Full. Skipping current generation.")
+
+    def multi_queue_it(self, count):
```

Comment:
Do we need this function if we just redirect to `single_queue_it`?

Reply:
I'm not sure I understand. Is it providing new functionality that the old process didn't have?